### PR TITLE
ci: scope github token for actions

### PR DESCRIPTION
**Pull request by aramase**
Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

**Reason for Change**:

By default `GITHUB_TOKEN` has write-all access. Scoping permissions to least required access.
https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token 

Once this PR is merged, we should also update the action > workflow permissions to read only (`Read repository contents permission`).

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [ ] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: .github/workflows/cover.yaml [hunk 1]
```diff
@@ -7,6 +7,9 @@ on:
     branches:
       - master
 
+permissions:
+  contents: write
```
[the remark this answers is not in the document]
Comment (aramase) [reply .github/workflows/cover.yaml:11]: Did some tests and setting permissions `contents: read` isn't sufficient for posting the status in the PR. Have to narrow down what's the exact set of permissions it requires.

[the remark this answers is not in the document]
Comment (nilekhc) [reply .github/workflows/cover.yaml]: @aramase Are you still working on this?

[the remark this answers is not in the document]
Comment (aramase) [reply .github/workflows/cover.yaml]: Yeah, will tag for review when it's ready

**Comment by nilekhc** [comment]
WIP

**Pull request closed without merging**